Add Flappy Bird Game

DavidSE12 opened the pull request:
Add 3 files including HTML, CSS and JavaScript for the Flappy Bird game
Input:
Player Input ( Space press ) will make the bird move upward
Environmental Input
+ Gravity: A constant downward force applied to the bird, simulating gravity
+ Obstacles: Obstacles appear on the screen at set interval

Output:
+ The bird flaps and moves up or down based on player control
+ The obstacles scroll on the screen
+ Game over will be displayed when player control bird hits the obstacle or hit rock bottom
+ Final score will be displayed on the screen

Process
Movement Mechanism
+ The bird move upward or downward based on player control
+ The gravity is continously applied to pull the bird downward
Collision Detection
+ The game checks if the bird collides with the pipes, ground of boundary.
Scrolling Environment
+ The obstacle move from right to left to simulate forward movement of the bird
Game Over condition
+ If the bird collides with any obstacle, the game transitions to the "Game Over" state.

URL: https://davidse12.github.io/FlappyBird/

DavidSE12 committed:
Add 3 files HTML, CSS and JS

DavidSE12 committed:
Update comment for every line code in main.js

DavidSE12 committed:
Update comment for every line of code in index.html

DavidSE12 committed:
Update comments on every line of codes in index.css

DavidSE12 committed:
Add Images

DavidSE12 committed:
Delete background1.png

DavidSE12 committed:
Delete bird3.png

DavidSE12 committed:
Create Images

DavidSE12 committed:
Delete Images

huythangphan commented on index.css:
```diff
@@ -0,0 +1,83 @@
+    position: absolute; /* Position the background relative to the interface */
+    top : 0;
+    left: 0;
+    height: 800px;
```
CS1: em sử dụng rem thay cho px mục đích là để sau này có tăng base font size thì các elements đều được scale theo

DavidSE12 replied:
Resolved

huythangphan commented on index.css:
```diff
@@ -0,0 +1,83 @@
+        right: 110vw; /* Move to the left beyond the viewport width */
+    }
+}
+
+
+
+
+
+
+
```
CS1: em remove unnecessary empty lines

DavidSE12 replied:
Resolved

huythangphan commented on main.js:
```diff
@@ -0,0 +1,88 @@
+// create variable to get elements from HTML file
+let block = document.getElementById("block");
```
CS1: a thấy variable này ko được sử dụng thì ko cần phải khai báo

DavidSE12 replied:
Resolved

huythangphan commented on main.js:
```diff
@@ -0,0 +1,88 @@
+// create variable to get elements from HTML file
+let block = document.getElementById("block");
+let hole = document.getElementById("hole");
+let bird = document.getElementById("bird");
```
CS1: a thấy 2 variables này ko thay đổi nên dùng let chưa hợp lý lắm

DavidSE12 replied:
Resolved

huythangphan commented on main.js:
```diff
@@ -0,0 +1,88 @@
+let block = document.getElementById("block");
+let hole = document.getElementById("hole");
+let bird = document.getElementById("bird");
+let jumping = 0;  // Flag to determine if the bird is jumping
```
CS5.1: em đặt tên biến phản ánh đúng data nó hold nhe

DavidSE12 replied:
Resolved

huythangphan commented on main.js:
```diff
@@ -0,0 +1,88 @@
+// Gravity function to simualate falling and detect collision for the game over
+setInterval( () => {
+    // Input: Get the current top distance of the bird (in px)
+    let birdtop = 
+    parseInt(window.getComputedStyle(bird).getPropertyValue("top")); // the value of top is a string so should change to interger
+
+    // Input: Get the current top and left distances of the hole
+    let holetop = 
+    parseInt(window.getComputedStyle(hole).getPropertyValue("top"));   
+    let holeleft = 
```
CS5.1: em đặt tên biến phản ánh đúng data nó hold nhe

DavidSE12 replied:
Resolved

huythangphan reviewed (changes requested):
em update thêm 1 vài thứ nữa nhe:
1. PR title có nghĩa hơn ví dụ `Add ...` hoặc `Update ...` hoặc `Fix bug ...`
2. bổ sung thêm PR description theo IPO
3. deploy lên github pages và gán link vào PR description để a test thử

DavidSE12 committed:
Add files via upload

DavidSE12 committed:
Create check

DavidSE12 committed:
Add files via upload

DavidSE12 committed:
Delete background1.png

DavidSE12 committed:
Delete bird3.png

DavidSE12 committed:
Update index.css

Update code in index.css file (change from "px" to "rem", delete line empty code)

DavidSE12 committed:
Update main.js

Change variable name that reflects their data

DavidSE12 committed:
Update main.js

Change variables name and remove empty line of codes

huythangphan commented on Images/check:
CS1: a thấy file này empty và ko được sử dụng do đó em nên remove nhe

huythangphan commented on main.js:
```diff
@@ -0,0 +1,87 @@
+        bird.style.top = (birdTopValue + 3) + "px";  // Increase the bird's top distance to simulate gravity
+    }
+
+    // Process: Check for game over conditions
```
CS1: logic này em nên tách ra 1 function riêng check game over sau này em muốn sửa logic lúc nào game over chỉ cần vào function đó thay vì tìm ở trong đoạn code dài như này

huythangphan commented on main.js:
```diff
@@ -0,0 +1,87 @@
+
+// Changing the position of hole after every animations
+hole.addEventListener('animationiteration' , () => {
+    // Process: Generate a random position for the hole's top between 200px and 450px
```
CS1: những thông tin như min/max hole top position, starting position of bird, game over threshold là constant do đó em nên define constant variables ở trên cùng sau này mình muốn sửa logic adjust các thông tin này thì chỉ cần sửa ở constant sẽ dễ hơn

huythangphan commented on main.js:
```diff
@@ -0,0 +1,87 @@
+    // Condition 2: Bird hits the hole (if the hole is within 300px to 190px horizontally and the bird is outside the hole's vertical range)
+    if(birdTopValue > 700 || (  (holeLeftValue < 300  && holeLeftValue > 190 ) && ( (birdTopValue < holeTopValue) || (birdTopValue > holeTopValue + 120)))){
+        // Output: Alert the game over message and display the score
+        alert("game over score " + (score - 1) )
```
CS1: em nên sử dụng template string để improve readability. sau này có chèn nhiều variable vào cũng sẽ dễ đọc hơn

huythangphan reviewed (changes requested):
PR title a thấy nen để là `Add flappy bird game` sau đó trong description đã có mô tả kỹ hơn là add những gì để đạt được `add flappy bird game` thì hợp lý hơn. em đang để là add files... thì a thấy nó hơi chung chung ko diễn tả được PR này thêm feature gì vào

DavidSE12 committed:
Delete Images/check

DavidSE12 committed:
Update main.js

- Separate condition check game over to outside function (easier to see, maintain and change in future) on line 21-28
- Using variable for min, max of hole instead of write value -> Easier to update in future

DavidSE12 committed:
Update main.js

Change name variable for min and max Hole Position appropriately

DavidSE12 committed:
Update main.js

Update the correct code from desktop
